### PR TITLE
add kubernetes-livenessprobe

### PR DESCRIPTION
**Pull request by joshrwolf**
image: add kubernetes-livenessprobe

Related: 

### Pre-review Checklist

- [ ] **IMPORTANT: 'image-request' tag has been applied if this PR is adding any images, including new versions or variants**

#### For new image PRs only

If you have an apko.yaml file in this PR you need to follow this checklist, otherwise feel free to remove.
- [ ] Include tests, sufficient enough that you would trust this image running in production.
- [ ] The version included is the latest GA version of the software
- [ ] The latest tag points to the newest stable version
- [ ] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] The image runs as `nonroot` and GID/UID are set to 65532
  - [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] ENTRYPOINT
  - [ ] For applications/servers/utilities call main program with no arguments e.g. [redis-server]
  - [ ] For base images leave empty
  - [ ] For dev variants set to entrypoint script that falls back to system
- [ ] CMD:
  - [ ] For server applications give arguments to start in daemon mode (may be empty)
  - [ ] For utilities/tooling bring up help e.g. `–help`
  - [ ] For base images with a shell, call it e.g. [/bin/sh]
- [ ] Consider where and how the image deviates from popular alternatives. Is there a good reason and is it documented?
- [ ] Add annotations e.g:
```
annotations:
  "org.opencontainers.image.authors": "Chainguard Team https://www.chainguard.dev/"
  "org.opencontainers.image.url": https://edu.chainguard.dev/chainguard/chainguard-images/reference/busybox/ # use the academy site here
  "org.opencontainers.image.source": https://github.com/chainguard-images/images/tree/main/images/bazel # use github here
```
- [ ] Check if environment variables are needed e.g. to set data locations
- [ ] Ensure the image responds to SIGTERM
  - [ ] `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`
- [ ] Documentation. Let's make this excellent. Include usage example.
- [ ] Error logs write to stderr and normal logs to stdout. DO NOT write to file.

### CODE REVIEW

File: images/kubernetes-livenessprobe/configs/latest.apko.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      uid: 65532
+      gid: 65532
+  run-as: 65532
+      
+entrypoint:
+  command: /usr/bin/livenessprobe
+    
+archs:
+  - x86_64
+  # - aarch64
```
Comment (imjasonh) [inline images/kubernetes-livenessprobe/configs/latest.apko.yaml:28]:
```suggestion
  - aarch64
```

File: images/kubernetes-livenessprobe/configs/latest.apko.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+contents:
+  repositories:
+    - /packages
```
Comment (imjasonh) [inline images/kubernetes-livenessprobe/configs/latest.apko.yaml:3]:
```suggestion
    - ./packages
```

File: images/kubernetes-livenessprobe/configs/latest.apko.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    - https://packages.wolfi.dev/os
+  keyring:
+    - /packages/local-melange.rsa.pub
```
Comment (imjasonh) [inline images/kubernetes-livenessprobe/configs/latest.apko.yaml:6]: I think we want this to be the chainguard signing pub key?

File: images/kubernetes-livenessprobe/configs/latest.apko.yaml (hunk 2 above)
Comment (developer-guy) [reply images/kubernetes-livenessprobe/configs/latest.apko.yaml]: I don't think that we need this

File: images/kubernetes-livenessprobe/configs/latest.apko.yaml (hunk 3 above)
Comment (developer-guy) [reply images/kubernetes-livenessprobe/configs/latest.apko.yaml]: I don't think that we need this either

**Comment by developer-guy** [comment]
@joshrwolf I think we need a real environment to test these liveness-probe and node-driver-registrar components since they didn't work in the docker environment it requires some sockets to be opened by other components, so, I removed these tests to be able to pass from checks fyi.

**Comment by developer-guy** [comment]
I think everything seems ready you can get the issue out of the draft.

File: images/kubernetes-csi-livenessprobe/image.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+status: experimental
```
Comment (imjasonh) [inline images/kubernetes-csi-livenessprobe/image.yaml:1]:
```suggestion
```

File: images/kubernetes-csi-livenessprobe/tests/01-help.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o errtrace -o pipefail -x
+
+if [[ "${IMAGE_NAME}" == "" ]]; then
+	echo "Must set IMAGE_NAME environment variable. Exiting."
+	exit 1
+fi
+
+docker run --rm "${IMAGE_NAME}" --help
```
Comment (imjasonh) [inline images/kubernetes-csi-livenessprobe/tests/01-help.sh:10]: Let's figure out what we can test in this environment, and document manual testing steps for what else we can't test here.

File: images/kubernetes-csi-node-driver-registrar/image.yaml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+status: experimental
```
Comment (imjasonh) [inline images/kubernetes-csi-node-driver-registrar/image.yaml:1]:
```suggestion
```

File: images/kubernetes-csi-node-driver-registrar/tests/01-version.sh [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o errtrace -o pipefail -x
+
+if [[ "${IMAGE_NAME}" == "" ]]; then
+	echo "Must set IMAGE_NAME environment variable. Exiting."
+	exit 1
+fi
+
+docker run --rm "${IMAGE_NAME}" --version
```
Comment (imjasonh) [inline images/kubernetes-csi-node-driver-registrar/tests/01-version.sh:10]: Same here

**Comment by mattmoor** [comment]
This seems close and is linked to 2 P0s, is it blocked on anything?

**Review by developer-guy — APPROVED** [review]

**Review by imjasonh — APPROVED** [review]

**Pull request merged**